unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendsController.cs

```diff
@@ -496,16 +532,25 @@ private void UpdateFriendshipStatus(FriendshipUpdateStatusMessage msg)
                 return;
 
             if (!friends.ContainsKey(userId))
+            {
                 friends.Add(userId, new UserStatus { userId = userId });
 
+                // Not adding the user to friendsSortedByName because this is only called in the legacy flow of a friendship
+            }
+
             if (ItsAnOutdatedUpdate(userId, friendshipStatus))
                 return;
 
             friends[userId].friendshipStatus = friendshipStatus;
 
             if (friendshipStatus == FriendshipStatus.NOT_FRIEND)
+            {
+                var friend = this.friends[userId];
                 friends.Remove(userId);
 
+                if (!string.IsNullOrEmpty(friend.userName) && friendsSortedByName.ContainsKey(friend.userName)) { friendsSortedByName.Remove(friend.userName); }
```

Comment:
no need to check if the key is exists.. apart from that, add a new line for better readability

Reply:
There's need for the tests in the old way of doing things, with the FF it's not needed